Thanks for addressing the issue. But it seems CI does not show the full log; the issue occurs in many places! Let me share the full warnings when I build locally. I am not sure that fixing only `hci_core.c` can resolve the issue. Please check this PR https://github.com/zephyrproject-rtos/zephyr/pull/86606 if you want to reproduce the issue

Build command: `west build -p auto -b rza3ul_smarc/r9a07g063u02gbg tests/bluetooth/classic/sdp_c -T sdp.c`

```
[216/324] Building C object zephyr/subsys/bluetooth/host/CMakeFiles/subsys__bluetooth__host.dir/hci_core.c.obj
In file included from /zephyr/include/zephyr/logging/log.h:11,
                 from /zephyr/include/zephyr/debug/stack.h:15,
                 from /zephyr/subsys/bluetooth/host/hci_core.c:25:
/zephyr/subsys/bluetooth/host/hci_core.c: In function 'hci_num_completed_packets':
/zephyr/subsys/bluetooth/host/hci_core.c:570:25: warning: format '%u' expects argument of type 'unsigned int', but argument 3 has type 'long unsigned int' [-Wformat=]
  570 |                 LOG_ERR("evt num_handles (=%u) too large (%u > %u)",
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  571 |                         evt->num_handles,
  572 |                         sizeof(*evt) + sizeof(evt->h[0]) * evt->num_handles,
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                      |
      |                                      long unsigned int
/zephyr/include/zephyr/logging/log_core.h:315:50: note: in definition of macro 'Z_LOG2'
  315 |                         z_log_printf_arg_checker(__VA_ARGS__);                                     \
      |                                                  ^~~~~~~~~~~
/zephyr/include/zephyr/logging/log.h:44:25: note: in expansion of macro 'Z_LOG'
   44 | #define LOG_ERR(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                         ^~~~~
/zephyr/subsys/bluetooth/host/hci_core.c:570:17: note: in expansion of macro 'LOG_ERR'
  570 |                 LOG_ERR("evt num_handles (=%u) too large (%u > %u)",
      |                 ^~~~~~~
/zephyr/subsys/bluetooth/host/hci_core.c:570:60: note: format string is defined here
  570 |                 LOG_ERR("evt num_handles (=%u) too large (%u > %u)",
      |                                                           ~^
      |                                                            |
      |                                                            unsigned int
      |                                                           %lu
[221/324] Building C object zephyr/subsys/bluetooth/host/classic/CMakeFiles/subsys__bluetooth__host__classic.dir/rfcomm.c.obj
In file included from /zephyr/include/zephyr/logging/log.h:11,
                 from /zephyr/include/zephyr/debug/stack.h:15,
                 from /zephyr/subsys/bluetooth/host/classic/rfcomm.c:15:
/zephyr/subsys/bluetooth/host/classic/rfcomm.c: In function 'rfcomm_recv':
/zephyr/subsys/bluetooth/host/classic/rfcomm.c:1548:25: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long unsigned int' [-Wformat=]
 1548 |                 LOG_ERR("Too small RFCOMM information (%d < %d)", buf->len,
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1549 |                         hdr_len + msg_len + sizeof(fcs));
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                           |
      |                                           long unsigned int
/zephyr/include/zephyr/logging/log_core.h:315:50: note: in definition of macro 'Z_LOG2'
  315 |                         z_log_printf_arg_checker(__VA_ARGS__);                                     \
      |                                                  ^~~~~~~~~~~
/zephyr/include/zephyr/logging/log.h:44:25: note: in expansion of macro 'Z_LOG'
   44 | #define LOG_ERR(...)    Z_LOG(LOG_LEVEL_ERR, __VA_ARGS__)
      |                         ^~~~~
/zephyr/subsys/bluetooth/host/classic/rfcomm.c:1548:17: note: in expansion of macro 'LOG_ERR'
 1548 |                 LOG_ERR("Too small RFCOMM information (%d < %d)", buf->len,
      |                 ^~~~~~~
/zephyr/subsys/bluetooth/host/classic/rfcomm.c:1548:62: note: format string is defined here
 1548 |                 LOG_ERR("Too small RFCOMM information (%d < %d)", buf->len,
      |                                                             ~^
      |                                                              |
      |                                                              int
      |                                                             %ld
[228/324] Building C object zephyr/subsys/bluetooth/host/CMakeFiles/subsys__bluetooth__host.dir/l2cap.c.obj
In file included from /zephyr/include/zephyr/logging/log.h:11,
                 from /zephyr/subsys/bluetooth/host/l2cap.c:35:
/zephyr/subsys/bluetooth/host/l2cap.c: In function 'l2cap_chan_le_recv_sdu':
/zephyr/subsys/bluetooth/host/l2cap.c:2453:17: warning: format '%zu' expects argument of type 'size_t', but argument 3 has type 'int' [-Wformat=]
 2453 |         LOG_DBG("chan %p len %zu", chan, buf->len);
      |                 ^~~~~~~~~~~~~~~~~        ~~~~~~~~
      |                                             |
      |                                             int
/zephyr/include/zephyr/logging/log_core.h:315:50: note: in definition of macro 'Z_LOG2'
  315 |                         z_log_printf_arg_checker(__VA_ARGS__);                                     \
      |                                                  ^~~~~~~~~~~
/zephyr/include/zephyr/logging/log.h:75:25: note: in expansion of macro 'Z_LOG'
   75 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/zephyr/subsys/bluetooth/host/l2cap.c:2453:9: note: in expansion of macro 'LOG_DBG'
 2453 |         LOG_DBG("chan %p len %zu", chan, buf->len);
      |         ^~~~~~~
/zephyr/subsys/bluetooth/host/l2cap.c:2453:32: note: format string is defined here
 2453 |         LOG_DBG("chan %p len %zu", chan, buf->len);
      |                              ~~^
      |                                |
      |                                long unsigned int
      |                              %u
/zephyr/subsys/bluetooth/host/l2cap.c: In function 'l2cap_chan_le_recv_seg':
/zephyr/subsys/bluetooth/host/l2cap.c:2495:17: warning: format '%zu' expects argument of type 'size_t', but argument 4 has type 'int' [-Wformat=]
 2495 |         LOG_DBG("chan %p seg %d len %zu", chan, seg, buf->len);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~             ~~~~~~~~
      |                                                         |
      |                                                         int
/zephyr/include/zephyr/logging/log_core.h:315:50: note: in definition of macro 'Z_LOG2'
  315 |                         z_log_printf_arg_checker(__VA_ARGS__);                                     \
      |                                                  ^~~~~~~~~~~
/zephyr/include/zephyr/logging/log.h:75:25: note: in expansion of macro 'Z_LOG'
   75 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/zephyr/subsys/bluetooth/host/l2cap.c:2495:9: note: in expansion of macro 'LOG_DBG'
 2495 |         LOG_DBG("chan %p seg %d len %zu", chan, seg, buf->len);
      |         ^~~~~~~
/zephyr/subsys/bluetooth/host/l2cap.c:2495:39: note: format string is defined here
 2495 |         LOG_DBG("chan %p seg %d len %zu", chan, seg, buf->len);
      |                                     ~~^
      |                                       |
      |                                       long unsigned int
      |                                     %u
/zephyr/subsys/bluetooth/host/l2cap.c: In function 'bt_l2cap_chan_send':
/zephyr/subsys/bluetooth/host/l2cap.c:3291:17: warning: format '%zu' expects argument of type 'size_t', but argument 4 has type 'int' [-Wformat=]
 3291 |         LOG_DBG("chan %p buf %p len %zu", chan, buf, buf->len);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~             ~~~~~~~~
      |                                                         |
      |                                                         int
/zephyr/include/zephyr/logging/log_core.h:315:50: note: in definition of macro 'Z_LOG2'
  315 |                         z_log_printf_arg_checker(__VA_ARGS__);                                     \
      |                                                  ^~~~~~~~~~~
/zephyr/include/zephyr/logging/log.h:75:25: note: in expansion of macro 'Z_LOG'
   75 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/zephyr/subsys/bluetooth/host/l2cap.c:3291:9: note: in expansion of macro 'LOG_DBG'
 3291 |         LOG_DBG("chan %p buf %p len %zu", chan, buf, buf->len);
      |         ^~~~~~~
/zephyr/subsys/bluetooth/host/l2cap.c:3291:39: note: format string is defined here
 3291 |         LOG_DBG("chan %p buf %p len %zu", chan, buf, buf->len);
      |                                     ~~^
      |                                       |
      |                                       long unsigned int
      |                                     %u
[235/324] Building C object zephyr/subsys/bluetooth/host/classic/CMakeFiles/subsys__bluetooth__host__classic.dir/l2cap_br.c.obj
In file included from /zephyr/include/zephyr/logging/log.h:11,
                 from /zephyr/subsys/bluetooth/host/classic/l2cap_br.c:33:
/zephyr/subsys/bluetooth/host/classic/l2cap_br.c: In function 'bt_l2cap_br_send_cb':
/zephyr/subsys/bluetooth/host/classic/l2cap_br.c:322:17: warning: format '%zu' expects argument of type 'size_t', but argument 4 has type 'int' [-Wformat=]
  322 |         LOG_DBG("chan %p buf %p len %zu", br_chan, buf, buf->len);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~                ~~~~~~~~
      |                                                            |
      |                                                            int
/zephyr/include/zephyr/logging/log_core.h:315:50: note: in definition of macro 'Z_LOG2'
  315 |                         z_log_printf_arg_checker(__VA_ARGS__);                                     \
      |                                                  ^~~~~~~~~~~
/zephyr/include/zephyr/logging/log.h:75:25: note: in expansion of macro 'Z_LOG'
   75 | #define LOG_DBG(...)    Z_LOG(LOG_LEVEL_DBG, __VA_ARGS__)
      |                         ^~~~~
/zephyr/subsys/bluetooth/host/classic/l2cap_br.c:322:9: note: in expansion of macro 'LOG_DBG'
  322 |         LOG_DBG("chan %p buf %p len %zu", br_chan, buf, buf->len);
      |         ^~~~~~~
/zephyr/subsys/bluetooth/host/classic/l2cap_br.c:322:39: note: format string is defined here
  322 |         LOG_DBG("chan %p buf %p len %zu", br_chan, buf, buf->len);
      |                                     ~~^
      |                                       |
      |                                       long unsigned int
      |                                     %u
[237/324] Building C object zephyr/subsys/bluetooth/host/shell/CMakeFiles/subsys__bluetooth__host__shell.dir/bt.c.obj
/zephyr/subsys/bluetooth/host/shell/bt.c: In function 'data_verbose_cb':
/zephyr/subsys/bluetooth/host/shell/bt.c:364:33: warning: field width specifier '*' expects argument of type 'int', but argument 2 has type 'size_t' {aka 'long unsigned int'} [-Wformat=]     
  364 |         bt_shell_fprintf_info("%*sType 0x%02x: ", strlen(scan_response_label), "", data->type);
      |                                ~^~                ~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                 |                 |
      |                                 int               size_t {aka long unsigned int}
In file included from /zephyr/subsys/bluetooth/host/shell/bt.c:41:
/zephyr/subsys/bluetooth/host/shell/bt.c: In function 'scan_recv':
/zephyr/subsys/bluetooth/host/shell/bt.c:574:31: warning: field width specifier '*' expects argument of type 'int', but argument 2 has type 'size_t' {aka 'long unsigned int'} [-Wformat=]     
  574 |                 bt_shell_info("%*s[SCAN DATA START - %s]",
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~
  575 |                               strlen(scan_response_label), "",
      |                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                               |
      |                               size_t {aka long unsigned int}
/zephyr/subsys/bluetooth/common/bt_shell_private.h:58:31: note: in definition of macro 'bt_shell_info'
   58 |         bt_shell_fprintf_info(_ft "\n", ##__VA_ARGS__)
      |                               ^~~
/zephyr/subsys/bluetooth/host/shell/bt.c:574:33: note: format string is defined here
  574 |                 bt_shell_info("%*s[SCAN DATA START - %s]",
      |                                ~^~
      |                                 |
      |                                 int
/zephyr/subsys/bluetooth/host/shell/bt.c:578:31: warning: field width specifier '*' expects argument of type 'int', but argument 2 has type 'size_t' {aka 'long unsigned int'} [-Wformat=]     
  578 |                 bt_shell_info("%*s[SCAN DATA END]", strlen(scan_response_label), "");
      |                               ^~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                     |
      |                                                     size_t {aka long unsigned int}
/zephyr/subsys/bluetooth/common/bt_shell_private.h:58:31: note: in definition of macro 'bt_shell_info'
   58 |         bt_shell_fprintf_info(_ft "\n", ##__VA_ARGS__)
      |                               ^~~
/zephyr/subsys/bluetooth/host/shell/bt.c:578:33: note: format string is defined here
  578 |                 bt_shell_info("%*s[SCAN DATA END]", strlen(scan_response_label), "");
      |                                ~^~
      |                                 |
      |                                 int
/zephyr/subsys/bluetooth/host/shell/bt.c: In function 'ad_init':
/zephyr/subsys/bluetooth/host/shell/bt.c:1938:40: warning: format '%d' expects argument of type 'int', but argument 2 has type 'ssize_t' {aka 'long int'} [-Wformat=]
 1938 |                         bt_shell_error("Failed to add CSIS data (err %d)", csis_ad_len);
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~
      |                                                                            |
      |                                                                            ssize_t {aka long int}
/zephyr/subsys/bluetooth/common/bt_shell_private.h:88:32: note: in definition of macro 'bt_shell_error'
   88 |         bt_shell_fprintf_error(_ft "\n", ##__VA_ARGS__)
      |                                ^~~
/zephyr/subsys/bluetooth/host/shell/bt.c:1938:71: note: format string is defined here
 1938 |                         bt_shell_error("Failed to add CSIS data (err %d)", csis_ad_len);
      |                                                                      ~^
      |                                                                       |
      |                                                                       int
      |                                                                      %ld
[244/324] Building C object zephyr/subsys/bluetooth/host/classic/CMakeFiles/subsys__bluetooth__host__classic.dir/sdp.c.obj
In file included from /zephyr/include/zephyr/logging/log.h:11,
                 from /zephyr/subsys/bluetooth/host/classic/sdp.c:28:
/zephyr/subsys/bluetooth/host/classic/sdp.c: In function 'get_att_search_list':
/zephyr/subsys/bluetooth/host/classic/sdp.c:1011:33: warning: format '%u' expects argument of type 'unsigned int', but argument 2 has type 'size_t' {aka 'long unsigned int'} [-Wformat=]      
 1011 |                         LOG_WRN("Exceeded maximum array length %u of %p", max_filters, filter);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~
      |                                                                           |
      |                                                                           size_t {aka long unsigned int}
/zephyr/include/zephyr/logging/log_core.h:315:50: note: in definition of macro 'Z_LOG2'
  315 |                         z_log_printf_arg_checker(__VA_ARGS__);                                     \
      |                                                  ^~~~~~~~~~~
/zephyr/include/zephyr/logging/log.h:55:24: note: in expansion of macro 'Z_LOG'
   55 | #define LOG_WRN(...)   Z_LOG(LOG_LEVEL_WRN, __VA_ARGS__)
      |                        ^~~~~
/zephyr/subsys/bluetooth/host/classic/sdp.c:1011:25: note: in expansion of macro 'LOG_WRN'
 1011 |                         LOG_WRN("Exceeded maximum array length %u of %p", max_filters, filter);
      |                         ^~~~~~~
/zephyr/subsys/bluetooth/host/classic/sdp.c:1011:65: note: format string is defined here
 1011 |                         LOG_WRN("Exceeded maximum array length %u of %p", max_filters, filter);
      |                                                                ~^
      |                                                                 |
      |                                                                 unsigned int
      |                                                                %lu

--- a/subsys/bluetooth/host/classic/l2cap_br.c
+++ b/subsys/bluetooth/host/classic/l2cap_br.c
@@ -319,7 +319,7 @@ int bt_l2cap_br_send_cb(struct bt_conn *conn, uint16_t cid, struct net_buf *buf,
 
 	br_chan = CONTAINER_OF(ch, struct bt_l2cap_br_chan, chan);
 
-	LOG_DBG("chan %p buf %p len %zu", br_chan, buf, buf->len);
+	LOG_DBG("chan %p buf %p len %u", br_chan, buf, buf->len);
 
 	hdr = net_buf_push(buf, sizeof(*hdr));
 	hdr->len = sys_cpu_to_le16(buf->len - sizeof(*hdr));

--- a/subsys/bluetooth/host/classic/rfcomm.c
+++ b/subsys/bluetooth/host/classic/rfcomm.c
@@ -1545,7 +1545,7 @@ static int rfcomm_recv(struct bt_l2cap_chan *chan, struct net_buf *buf)
 	}
 
 	if (buf->len < (hdr_len + msg_len + sizeof(fcs))) {
-		LOG_ERR("Too small RFCOMM information (%d < %d)", buf->len,
+		LOG_ERR("Too small RFCOMM information (%u < %zu)", buf->len,
 			hdr_len + msg_len + sizeof(fcs));
 		return 0;
 	}

--- a/subsys/bluetooth/host/classic/sdp.c
+++ b/subsys/bluetooth/host/classic/sdp.c
@@ -1008,7 +1008,7 @@ static uint16_t get_att_search_list(struct net_buf *buf, uint32_t *filter,
 
 	while (size) {
 		if (*num_filters >= max_filters) {
-			LOG_WRN("Exceeded maximum array length %u of %p", max_filters, filter);
+			LOG_WRN("Exceeded maximum array length %zu of %p", max_filters, filter);
 			return 0;
 		}
 

--- a/subsys/bluetooth/host/hci_core.c
+++ b/subsys/bluetooth/host/hci_core.c
@@ -567,7 +567,7 @@ static void hci_num_completed_packets(struct net_buf *buf)
 	int i;
 
 	if (sizeof(*evt) + sizeof(evt->h[0]) * evt->num_handles > buf->len) {
-		LOG_ERR("evt num_handles (=%u) too large (%u > %u)",
+		LOG_ERR("evt num_handles (=%u) too large (%zu > %u)",
 			evt->num_handles,
 			sizeof(*evt) + sizeof(evt->h[0]) * evt->num_handles,
 			buf->len);

--- a/subsys/bluetooth/host/l2cap.c
+++ b/subsys/bluetooth/host/l2cap.c
@@ -2450,7 +2450,7 @@ static void l2cap_chan_le_recv_sdu(struct bt_l2cap_le_chan *chan,
 {
 	int err;
 
-	LOG_DBG("chan %p len %zu", chan, buf->len);
+	LOG_DBG("chan %p len %u", chan, buf->len);
 
 	__ASSERT_NO_MSG(bt_l2cap_chan_get_state(&chan->chan) == BT_L2CAP_CONNECTED);
 	__ASSERT_NO_MSG(atomic_get(&chan->rx.credits) == 0);
@@ -2492,7 +2492,7 @@ static void l2cap_chan_le_recv_seg(struct bt_l2cap_le_chan *chan,
 	/* Store received segments in user_data */
 	memcpy(net_buf_user_data(chan->_sdu), &seg, sizeof(seg));
 
-	LOG_DBG("chan %p seg %d len %zu", chan, seg, buf->len);
+	LOG_DBG("chan %p seg %d len %u", chan, seg, buf->len);
 
 	/* Append received segment to SDU */
 	len = net_buf_append_bytes(chan->_sdu, buf->len, buf->data, K_NO_WAIT,
@@ -3288,7 +3288,7 @@ int bt_l2cap_chan_send(struct bt_l2cap_chan *chan, struct net_buf *buf)
 		return -EINVAL;
 	}
 
-	LOG_DBG("chan %p buf %p len %zu", chan, buf, buf->len);
+	LOG_DBG("chan %p buf %p len %u", chan, buf, buf->len);
 
 	if (buf->ref != 1) {
 		LOG_WRN("Expecting 1 ref, got %d", buf->ref);

--- a/subsys/bluetooth/host/shell/bt.c
+++ b/subsys/bluetooth/host/shell/bt.c
@@ -361,7 +361,8 @@ static void print_data_set(uint8_t set_value_len,
 
 static bool data_verbose_cb(struct bt_data *data, void *user_data)
 {
-	bt_shell_fprintf_info("%*sType 0x%02x: ", strlen(scan_response_label), "", data->type);
+	bt_shell_fprintf_info("%*sType 0x%02x: ",
+			      (int)strlen(scan_response_label), "", data->type);
 
 	switch (data->type) {
 	case BT_DATA_UUID16_SOME:
@@ -572,10 +573,10 @@ static void scan_recv(const struct bt_le_scan_recv_info *info, struct net_buf_si
 
 	if (scan_verbose_output) {
 		bt_shell_info("%*s[SCAN DATA START - %s]",
-			      strlen(scan_response_label), "",
+			      (int)strlen(scan_response_label), "",
 			      scan_response_type_txt(info->adv_type));
 		bt_data_parse(&buf_copy, data_verbose_cb, NULL);
-		bt_shell_info("%*s[SCAN DATA END]", strlen(scan_response_label), "");
+		bt_shell_info("%*s[SCAN DATA END]", (int)strlen(scan_response_label), "");
 	}
 
 #if defined(CONFIG_BT_CENTRAL)
@@ -1935,7 +1936,7 @@ static ssize_t ad_init(struct bt_data *data_array, const size_t data_array_size,
 		csis_ad_len = csis_ad_data_add(&data_array[ad_len],
 					       data_array_size - ad_len, discoverable);
 		if (csis_ad_len < 0) {
-			bt_shell_error("Failed to add CSIS data (err %d)", csis_ad_len);
+			bt_shell_error("Failed to add CSIS data (err %zd)", csis_ad_len);
 			return ad_len;
 		}
 

